src/DynamoCore/Library/LibraryServices.cs

```diff
@@ -699,7 +699,16 @@ private bool TryGetDefaultArgumentFromAttribute(ArgumentInfo arg, out Associativ
             LibraryManagementCore.ParsingMode = ProtoCore.ParseMode.AllowNonAssignment;
             LibraryManagementCore.IsParsingCodeBlockNode = true;
 
-            var astNode = ParserUtils.ParseWithCore(defaultExpression + ";", LibraryManagementCore);
+            ProtoCore.AST.Node astNode = null ;
+            try
+            {
+                astNode = ParserUtils.ParseWithCore(defaultExpression + ";", LibraryManagementCore);
+            }
+            catch (ProtoCore.BuildHaltException ex)
+            {
+                Log("Failed to parse default argument attribute \"" + defaultExpression + "\" for parameter: " + arg.Name);
```

Comment:
For Log, I don't think it is necessary, but we have a [user story](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6261) about how to show error/warning for the compilation of default argument. For the time being, just log to console.